### PR TITLE
TenantDetail and TenantDelete handlers

### PR DESCRIPTION
**Pull request by daniellemaxwell**
### Scope of changes

Adds server-side handlers for TenantDetail and TenantDelete along with corresponding tests. 
`TenantDetail` will allow a user to pass in a given tenant id and get a summary detail for the specified tenant.
`TenantDelete` will allow a user to pass in a given tenant id and delete a tenant.

The complexity for this PR is low. Comments to the `TenantCreate` handler and test have been added as well.

Fixes SC-10278

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [x] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

Describe how reviewers can test this change to be sure that it works correctly. Add a checklist if possible.

### Author checklist

- [x] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [x]  I have added new test fixtures as needed to support added tests
- [ ]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.

### CODE REVIEW

File: pkg/tenant/tenants.go [hunk 1]
```diff
@@ -4,12 +4,20 @@ import (
...
 
 	"github.com/gin-gonic/gin"
+	"github.com/oklog/ulid/v2"
 	"github.com/rotationalio/ensign/pkg/tenant/api/v1"
 	"github.com/rotationalio/ensign/pkg/tenant/db"
 	"github.com/rs/zerolog/log"
 )
 
-// TenantCreates adds a new tenant to the database
+func (s *Server) TenantList(c *gin.Context) {
```
Comment (daniellemaxwell) [inline pkg/tenant/tenants.go:13]: This handler will be implemented via a separate PR.

File: pkg/tenant/tenants.go [hunk 2]
```diff
@@ -60,19 +75,65 @@ func (s *Server) TenantCreate(c *gin.Context) {
...
+	if err = db.DeleteTenant(c.Request.Context(), tenantID); err != nil {
+		log.Error().Err(err).Str("tenantID", tenantID.String()).Msg("could not delete tenant")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not delete tenant"))
+		return
+	}
+
+	c.JSON(http.StatusOK, tenant)
+}
+
+func (s *Server) TenantUpdate(c *gin.Context) {
```
Comment (daniellemaxwell) [inline pkg/tenant/tenants.go:137]: This handler will be implemented via a separate handler and this was only a position change.

File: pkg/tenant/tenants_test.go [hunk 3]
```diff
@@ -14,19 +16,78 @@ func (suite *tenantTestSuite) TestTenantCreate() {
...
+
+	// Connect to a mock trtl database
+	trtl := db.GetMock()
+	defer trtl.Reset()
+
+	// Get JSON test data.
+	data, err := os.ReadFile("testdata/tenant.json")
+	if err != nil {
+		return
+	}
```
Comment (pdeziel) [inline pkg/tenant/tenants_test.go:54]: Can we change this to a `require.NoError`? I think as the test is written currently, if we get an error here we will just return out of the test and the rest of the test code is skipped, which makes it look like the test has passed.

File: pkg/tenant/tenants_test.go [hunk 4]
```diff
@@ -14,19 +16,78 @@ func (suite *tenantTestSuite) TestTenantCreate() {
...
+	// Create a tenant test fixture.
+	req := &api.Tenant{
+		ID:              "01ARZ3NDEKTSV4RRFFQ69G5FAV",
+		Name:            "example-staging",
+		EnvironmentType: "prod",
+	}
+
+	tenant, err := suite.client.TenantDetail(ctx, tenantID)
+	require.Error(err, http.StatusBadRequest, "could not get tenant")
+	require.Equal(req, tenant, "tenant should match")
```
Comment (pdeziel) [inline pkg/tenant/tenants_test.go:72]: Can we add a few quick cases here and in `TestTenantDelete` to cover the error cases? We just need to be sure that if the required params are not supplied we're getting the correct HTTP status response back with an error. We should also test that 404s are returned when the trtl mock is set up to return an error.

**Review by pdeziel — DISMISSED** [review]
@daniellemaxwell This is a great start and the tenant API definitely makes sense at a high level. I did have some comments about the HTTP status codes we are returning but hopefully those are quick fixes. We probably also need to add a bit more test coverage as I've indicated in the comments to assert that we are returning the correct responses. Let me know if you want to discuss!

**Review by pdeziel — COMMENTED** [review]
Thanks for making the changes! I'll let @bbengfort have a chance to look at this before merging.

File: pkg/tenant/tenants.go [hunk 5]
```diff
@@ -1,15 +1,24 @@
...
 
-// TenantCreates adds a new tenant to the database
+func (s *Server) TenantList(c *gin.Context) {
+	c.JSON(http.StatusNotImplemented, "not implemented yet")
+}
+
+// TenantCreate adds a new tenant to the database
+// and returns a 201 StatusCreated response.
+//
+// Route: /tenant
```
Comment (bbengfort) [inline pkg/tenant/tenants.go:21]: I like the doc strings :-)

File: pkg/tenant/tenants.go [hunk 6]
```diff
@@ -18,22 +27,30 @@ func (s *Server) TenantCreate(c *gin.Context) {
...
 	if err = c.BindJSON(&t); err != nil {
 		log.Warn().Err(err).Msg("could not bind tenant create request")
 		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not bind request"))
 		return
 	}
 
+	// Verify that a tenant ID does not exist and return a 400 response if the
+	// tenant id exists.
 	if t.ID != "" {
 		c.JSON(http.StatusBadRequest, api.ErrorResponse("tenant id cannot be speicified on create"))
```
Comment (bbengfort) [inline pkg/tenant/tenants.go:41]:
```suggestion
		c.JSON(http.StatusBadRequest, api.ErrorResponse("tenant id cannot be specified on create"))
```

File: pkg/tenant/tenants.go [hunk 7]
```diff
@@ -60,19 +76,70 @@ func (s *Server) TenantCreate(c *gin.Context) {
...
 
-func (s *Server) TenantDetail(c *gin.Context) {
-	c.JSON(http.StatusNotImplemented, "not implemented yet")
-}
+	// Get the specified tenant from the database and return a 500 response
+	// if it can not be retrieved.
+	var tenant *db.Tenant
+	if tenant, err = db.RetrieveTenant(c.Request.Context(), tenantID); err != nil {
+		log.Error().Err(err).Msg("could not retrieve tenant")
+		fmt.Println(err.Error())
```
Comment (bbengfort) [inline pkg/tenant/tenants.go:103]:
```suggestion
```

Stray debugging line!

**Review by bbengfort — APPROVED** [review]
Looks great - thank you @pdeziel for the initial review and thank you @daniellemaxwell for sticking with the PR!

**Pull request merged**